File: TJ Code pre Fall-2026/ProgrammingTeam/Hailstone/Hailstone.py
sum = 0
num = int(input())
def myHailstone(num:int) -> int:
    global sum
    if (num == 1):
        sum += num
    else:
        if (num % 2 == 1):
            sum += num
            return myHailstone(3 * num + 1)
        if (num % 2 == 0):
            sum += num
            return myHailstone(num // 2)

myHailstone(num)
print(sum)

    

File: TJ Code pre Fall-2026/ProgrammingTeam/Hailstone/Hailstone.java
import java.util.*;
public class Hailstone {
    private static long sum = 0;
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        long num = sc.nextLong();
        hailstone(num);
        System.out.println(sum);

    }
    public static void hailstone(long num)
    {
        sum+= num;
        if(num != 1)
        {
            if (num % 2 == 0)
            {
                hailstone(num / 2);
            }
            else
            {
                hailstone((3 * num) + 1);
            }
        }

    }

}
